resources/analysisTools/qcPipeline/bashLib.sh

```diff
@@ -142,6 +142,18 @@ isDebugSet() {
     normalizeBoolean "${debug:-false}"
 }
 
+# Grep returns 1 if no match is found. Ensure that empty files are handled gracefully without exit due to `set -e`.
+grepIgnoreEmpty() {
+    (set +e    # The parentheses keep the `set -e` in the local subshell.
+    grep "$@"
+    if [[ $? -gt 1 ]]; then
```

Comment:
Afaik, the return code can only be in the range of [0;255] and therefore never be below zero and it is safe to just return $?.

Reply:
Yes it is totally good what you do, you could just have it shorter:
return $?